src/applications/accredited-representative-portal/utilities/api.js

```diff
@@ -1,40 +1,75 @@
+import merge from 'lodash/merge';
+import { fetchAndUpdateSessionExpiration } from 'platform/utilities/api';
+import environment from 'platform/utilities/environment';
+import localStorage from 'platform/utilities/storage/localStorage';
 import manifest from '../manifest.json';
 
+// Set app name for request headers
 window.appName = manifest.entryName;
 
 const API_VERSION = 'accredited_representative_portal/v0';
 
+const isJson = response => {
+  const contentType = response.headers.get('Content-Type');
+  return contentType && contentType.includes('application/json');
+};
+
 /**
- * This abstraction was introduced to let us pass fetch options to API methods
- * so that we could forward an abort signal from route loaders. This abstraction
- * only accomodates inner functions that don't have default parameters.
- *
- * Not every API method needs to be defined using this abstraction. Furthermore,
- * it is okay to refactor this abstraction, or even just unwind it altogether,
- * if that seems justified.
+ * Enhanced API wrapper that preserves Response objects for error handling
+ * while maintaining existing platform functionality where beneficial
  */
 const wrapApiRequest = fn => {
-  return (...args) => {
+  return async (...args) => {
+    // Set up request options similarly to platform apiRequest
     const optionsFromCaller = args[fn.length] || {};
     const [resource, optionsFromFn = {}] = fn(...args.slice(0, fn.length));
 
-    const options = {
-      apiVersion: API_VERSION,
+    const csrfTokenStored = localStorage.getItem('csrfToken');
+
+    const defaultSettings = {
+      method: 'GET',
+      credentials: 'include',
       headers: {
+        'X-Key-Inflection': 'camel',
+        'Source-App-Name': window.appName,
+        'X-CSRF-Token': csrfTokenStored,
         'Content-Type': 'application/json',
       },
-      ...optionsFromFn,
-      ...optionsFromCaller,
     };
 
-    return apiRequest(resource, options);
+    const settings = merge(defaultSettings, optionsFromFn, optionsFromCaller);
+
+    // Build URL like platform apiRequest
+    const baseUrl = `${environment.API_URL}/${API_VERSION}`;
+    const url = resource[0] === '/' ? [baseUrl, resource].join('') : resource;
+
+    try {
+      // Reuse session management from platform
+      const response = await fetchAndUpdateSessionExpiration(url, settings);
+
+      // Handle CSRF token updates
+      const csrfToken = response.headers.get('X-CSRF-Token');
+      if (csrfToken && csrfToken !== csrfTokenStored) {
+        localStorage.setItem('csrfToken', csrfToken);
+      }
+
+      // For successful responses, parse and return data
+      if (response.ok || response.status === 304) {
+        return isJson(response) ? await response.json() : response;
+      }
+
+      // For errors, preserve the Response object
+      throw response;
+    } catch (err) {
+      // Log network-like errors to Sentry
+      Sentry.withScope(scope => {
```

Comment:
- Incorrectly calling Sentry with thrown response objects?
  - Platform wasn't doing this